File: FindTheDifferenceOfTwoArrays/FindTheDifferenceOfTwoArrays.py
from typing import List

class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        s1, s2 = set(nums1), set(nums2)
        
        return [list(s1 - s2), list(s2 - s1)]

def main():
    sol = Solution()

    # Example 1
    nums1_1 = [1, 2, 3]
    nums2_1 = [2, 4, 6]
    print(f"Example 1: {sol.findDifference(nums1_1, nums2_1)}")  # Expected: [[1, 3], [4, 6]]

    # Example 2
    nums1_2 = [1, 2, 3, 3]
    nums2_2 = [1, 1, 2, 2]
    print(f"Example 2: {sol.findDifference(nums1_2, nums2_2)}")  # Expected: [[3], []]

if __name__ == "__main__":
    main()

File: FindTheDifferenceOfTwoArrays/FindTheDifferenceOfTwoArrays.rs
use std::collections::HashSet;

struct Solution;

impl Solution {
    pub fn find_difference(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<Vec<i32>> {
        let s1: HashSet<i32> = nums1.into_iter().collect();
        let s2: HashSet<i32> = nums2.into_iter().collect();

        vec![
            s1.difference(&s2).cloned().collect(),
            s2.difference(&s1).cloned().collect(),
        ]
    }
}

fn main() {
    let n1_1 = vec![1, 2, 3];
    let n1_2 = vec![2, 4, 6];
    println!("{:?}", Solution::find_difference(n1_1, n1_2));

    let n2_1 = vec![1, 2, 3, 3];
    let n2_2 = vec![1, 1, 2, 2];
    println!("{:?}", Solution::find_difference(n2_1, n2_2));
}